compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs

```diff
@@ -81,6 +81,20 @@ createProjectPackageDb opts thisSdkVer deps dataDeps = do
     createDirectoryIfMissing True $ dbPath </> "package.conf.d"
 
     deps <- expandSdkPackages (filter (`notElem` basePackages) deps)
+
+    let versionize name = case name of
+          "daml-script" -> name <> "-" <> sdkVersion
+          "daml-trigger" -> name <> "-" <> sdkVersion
+          _ -> name
+
+    let units =
+          [ GHC.stringToUnitId (versionize base)
+          | dep <- deps
+          , let file = takeFileName dep
```

Comment:
You can’t rely on the file names for figuring out the unit ids since these can be overwritten via `daml build -o`. We have a list of packages and their unit ids below at https://github.com/digital-asset/daml/blob/fcab7d1a0fa4c3ebd2a330519ebb57b1636382aa/compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs#L128 (note that this includes transitive dependencies so you don’t want all of those but you can use it to get the actual unit ids).

Reply:
we don't use the names any more